src/controller/CHIPDevice.cpp

```diff
@@ -565,7 +565,7 @@ void Device::OnSessionEstablishmentError(CHIP_ERROR error)
 
 void Device::OnSessionEstablished()
 {
-    mCASESession.PeerConnection().SetPeerNodeId(mDeviceId);
+    mCASESession.SetPeerNodeId(mDeviceId);
```

Comment:
This is kinda bizarre, in that the session should know which peer we are trying to connect to when we start trying to connect, not when the connection is established!  Pre-existing, but please file a followup to sort out what needs to happen to remove this line.

Reply:
Added a TODO tag.